kong/plugins/ip-restriction/handler.lua

```diff
@@ -2,10 +2,48 @@ local BasePlugin = require "kong.plugins.base_plugin"
 local responses = require "kong.tools.responses"
 local iputils = require "resty.iputils"
 
+
+local ok, new_tab = pcall(require, "table.new")
+if not ok then
+  new_tab = function() return {} end
+end
```

Comment:
Usually, we prefer defining those as:

```lua
local new_tab
do
  local ok
  ok, new_tab = pcall(require, "table.new")
  if not ok then
    new_tab = function() return {} end
  end
end
```

So that `ok` is not hanging in this module's top chunk level. This really should be a helper part of `kong.tools.public`.